### DISCUSSION TITLE
Welcome to firebase-admin-node Discussions!

### DISCUSSION THREAD

**Discussion by hiranya911** [firebase/firebase-admin-node#1229] [General]
## 👋 Welcome!
  We’re using Discussions as a place to get feedback on our new APIs and have high-level conversations that don't belong in Issues.

You can:
  * Share ideas.
  * Engage with other community members.
  * Give us open-ended feedback about our SDK.

There are only two rules:

1. **Welcome others and be open-minded.** Firebase is a community!
1. **Stay on topic.** This repository is only for the Firebase Admin Node.js SDK and is not the place to discuss general Firebase issues (StackOverflow, Reddit, etc are better forums for that)

**Comment by CrlsMrls** [comment]
I would like to propose a change when fetching a user with Firebase auth.
Currently it throws an error if the user is not found, the error code value is the string `'auth/user-not-found'`,
It would be great to have an `undefined` instead, as currently if you want to find out if a user exists, you need to catch up the error.

More concretely, instead of this code:

```
 async updateOrCreateUser(someDto) {
  const isUserInFirebase: boolean = await doesUserExist(dto.email);
  if (isUserInFirebase ) {
       // update the user ....
   } else {
      // create the user ....
   }
 }


  async doesUserExist(email: string): Promise<boolean> {
    try {
      await firebaseAdmin.auth().getUserByEmail(email);
      return true;
    } catch (error) {
      if (error.code !== 'auth/user-not-found') {
        // there was an error, not just the user was not found
        throw new InternalServerErrorException(error.message);
      }
    }
    return false;
  }
```

it would be much simpler

```
 async updateOrCreateUser(someDto) {
  const user: UserRecord | undefined = await firebaseAdmin.auth().getUserByEmail(email);;
  if (user ) {
       // update the user ....
   } else {
      // create the user ....
   }
 }

```

**Reply by hiranya911** [reply]
Can you start a new thread (or even an issue) for this please? I will remove this post once this feedback is recorded as a separate thread.
